### ISSUE TITLE
WX Phoenix AttributeError: 'NavigationToolbar2WxAgg' issue

### ISSUE THREAD

**Issue by jenshnielsen** [matplotlib/matplotlib#4460]
Trying to test the new WX Phonix backend under Python3 I get the following error:

```
Traceback (most recent call last):
  File "testwx.py", line 4, in <module>
    plt.plot(range(100))
  File "/usr/local/lib/python3.4/site-packages/matplotlib-1.5.dev1-py3.4-macosx-10.10-x86_64.egg/matplotlib/pyplot.py", line 3095, in plot
    ax = gca()
  File "/usr/local/lib/python3.4/site-packages/matplotlib-1.5.dev1-py3.4-macosx-10.10-x86_64.egg/matplotlib/pyplot.py", line 830, in gca
    ax =  gcf().gca(**kwargs)
  File "/usr/local/lib/python3.4/site-packages/matplotlib-1.5.dev1-py3.4-macosx-10.10-x86_64.egg/matplotlib/pyplot.py", line 463, in gcf
    return figure()
  File "/usr/local/lib/python3.4/site-packages/matplotlib-1.5.dev1-py3.4-macosx-10.10-x86_64.egg/matplotlib/pyplot.py", line 436, in figure
    **kwargs)
  File "/usr/local/lib/python3.4/site-packages/matplotlib-1.5.dev1-py3.4-macosx-10.10-x86_64.egg/matplotlib/backends/backend_wxagg.py", line 113, in new_figure_manager
    return new_figure_manager_given_figure(num, fig)
  File "/usr/local/lib/python3.4/site-packages/matplotlib-1.5.dev1-py3.4-macosx-10.10-x86_64.egg/matplotlib/backends/backend_wxagg.py", line 120, in new_figure_manager_given_figure
    frame = FigureFrameWxAgg(num, figure)
  File "/usr/local/lib/python3.4/site-packages/matplotlib-1.5.dev1-py3.4-macosx-10.10-x86_64.egg/matplotlib/backends/backend_wx.py", line 1274, in __init__
    tw, th = self.toolbar.GetSizeTuple()
AttributeError: 'NavigationToolbar2WxAgg' object has no attribute 'GetSizeTuple'
```

From my limited debugging it seems like GetSizeTuple is no more and we should use GetSize and unpack it manually.

Im using wxPython-Phoenix (3.0.3.dev1820+49a8884) on python 3.4.3 (Homebrew OSX 10.10)

**Comment by jenshnielsen** [comment]
By the way this is doing the simplest plot that I can think of 

```python
import matplotlib
matplotlib.use('WXAgg')
import matplotlib.pyplot as plt
plt.plot(range(100))
plt.show()
```

**Comment by WeatherGod** [comment]
attn: @wernerfb

**Comment by wernerfb** [comment]
mpl.pyplot uses FigureFrameWx which isn't used by the embedding_in_wx* samples.

After the following change the sample code runs for me on Windows, but no plot is shown, will try to work on this some more next week.

A quick hack (as I don't see how to get that into wx_compat) for this is to make the following change around line 1274:
            if wxc.is_phoenix:
                tw, th = self.toolbar.GetSize()
                fw, fh = self.canvas.GetSize()
            else:
                tw, th = self.toolbar.GetSizeTuple()
                fw, fh = self.canvas.GetSizeTuple()

**Comment by wernerfb** [comment]
With Py2.7 and 3.0.2 classic I see a crash with no traceback when I run it from the command line, when I run it against Phoenix I see this traceback:

Error in atexit._run_exitfuncs:
Traceback (most recent call last):
  File "c:\python27\lib\atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "c:\python27\lib\site-packages\matplotlib\_pylab_helpers.py", line 89, in destroy_all
    manager.destroy()
  File "c:\python27\lib\site-packages\matplotlib\backends\backend_wx.py", line 1376, in destroy
    self.frame.Destroy()
  File "c:\python27\lib\site-packages\matplotlib\backends\backend_wx.py", line 1334, in Destroy
    wx.Frame.Destroy(self, *args, **kwargs)
RuntimeError: wrapped C/C++ object of type FigureFrameWxAgg has been deleted
Error in sys.exitfunc:
Traceback (most recent call last):
  File "c:\python27\lib\atexit.py", line 24, in _run_exitfuncs
    func(*targs, **kargs)
  File "c:\python27\lib\site-packages\matplotlib\_pylab_helpers.py", line 89, in destroy_all
    manager.destroy()
  File "c:\python27\lib\site-packages\matplotlib\backends\backend_wx.py", line 1376, in destroy
    self.frame.Destroy()
  File "c:\python27\lib\site-packages\matplotlib\backends\backend_wx.py", line 1334, in Destroy
    wx.Frame.Destroy(self, *args, **kwargs)
RuntimeError: wrapped C/C++ object of type FigureFrameWxAgg has been deleted

Don't understand what mpl.pyplot does here, why destroy the frame?

Hope someone who does use mpl.pyplot with wx can jump in and help on this.

**Comment by WeatherGod** [comment]
Chances are, for some reason there is a different exception being thrown,
causing python to begin the exit process, and then that hits another bug
during the exit process and you don't see the first exception. Perhaps a
try/except around "show()" with some print messages might be illuminating?

**Comment by wernerfb** [comment]
Will give that a try, will also have to figure out how the wx app is created when one uses mpl.pyplot, hopefully I will find time sometimes towards the end of next week.

**Comment by jenshnielsen** [comment]
closed by #4479
